File: android-native-app/app/src/main/java/pl/edu/pb/androidnativeapp/data/infinitelist/model/CharacterEntity.kt
package pl.edu.pb.androidnativeapp.data.infinitelist.model

import com.squareup.moshi.Json

data class CharacterEntity(
    @field:Json(name = "id") val id: String,
    @field:Json(name = "name") val name: String,
    @field:Json(name = "status") val status: String,
    @field:Json(name = "species") val species: String,
    @field:Json(name = "gender") val gender: String,
    @field:Json(name = "image") val image: String,
)

File: flutter-multiplatform-app/flutter_multiplatform_app/lib/data/model/character_entity.dart
final class CharacterEntity {
  const CharacterEntity({
    required this.id,
    required this.name,
    required this.status,
    required this.species,
    required this.gender,
    required this.image,
  });

  final int id;
  final String name;
  final String status;
  final String species;
  final String gender;
  final String image;

  CharacterEntity.fromJson(Map<String, dynamic> json)
      : id = json['id'] as int,
        name = json['name'] as String,
        status = json['status'] as String,
        species = json['species'] as String,
        gender = json['gender'] as String,
        image = json['image'] as String;

  // Convert a CharacterEntity into a Map. The keys must correspond to the names of the columns in the database.
  Map<String, Object?> toMap() {
    return {
      'id': id,
      'name': name,
      'status': status,
      'species': species,
      'gender': gender,
      'image': image,
    };
  }
}
